torch_xla/experimental/fori_loop.py

```diff
@@ -94,6 +153,12 @@ def _xla_while_loop(cond_fn, body_fn, *carried_inputs, additional_inputs):
 
   # gain final result with generated while xlacomputation
   result = torch_xla._XLAC._xla_user_computation('xla::_op_test_while',
-                                                 (carried_inputs), computation)
+                                                 (total_inputs), computation)
+
+  # unwrapper result without additional_inputs for original order
+  additional_inputs_len = len(additional_inputs) + 1
+  final_res = [
+      result[0],
+  ] + result[additional_inputs_len:]
```

Comment:
why couldn't you just return `res`?

Reply:
thanks, we want to hide `weights/bias` like/match `torch._higher_order_ops.while_loop`, updated with comments;

---
### detail

[`xla::While`](https://github.com/pytorch/xla/blob/5c7d629ea6b8d79d625f7cb1afc4bce64bd0c77c/torch_xla/experimental/fori_loop.py#L155) return `(iteri, weights, bias, y)`;

then the final result user get from `torch._higher_order_ops, while_loop` would only contain `(iteri, ?)`, the rest has been cut to hide `weights/bias` from user;

so we reorder the result from `(iteri, weights, bias, y)` to `(iteri, y, weights, bias)`, so that user could get result `(iteri, y)` with `torch._higher_order_ops, while_loop` on XLA too

```
# user
iteri, y = torch._higher_order_ops.while_loop(cond_fn, body_fn, (iteri, x))

# pytorch/xla/torch_xla/experimental/fori_loop:_xla_while_loop()
iteri, weights, bias, y = _xla_while_loop(cond_fn, body_fn, (iteri, weights, bias, x))
```